### PR TITLE
new package: Seqan (bioinformatics)

### PR DESCRIPTION
**Pull request by Sinan81**
create new package

### CODE REVIEW

File: var/spack/repos/builtin/packages/seqan/package.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    performance, generality, extensibility, and integration with other
+    libraries. SeqAn is easy to use and simplifies the development of new
+    software tools with a minimal loss of performance"""
+
+    homepage = "https://www.seqan.de"
+    url      = "https://github.com/seqan/seqan/archive/seqan-v2.4.0.tar.gz"
+
+    version('2.4.0rc2', 'bd62772b3661bf07dad2bf8f69a1bdaf')
+
+    #depends on C++14 compiler: gcc >= 5, Clang/LLVM >= 3.6, Icc >= 17
```
Comment (davydden) [inline var/spack/repos/builtin/packages/seqan/package.py]: you can add all that into conflicts

Comment (Sinan81) [reply var/spack/repos/builtin/packages/seqan/package.py]: Thanks! I was indeed wondering how to to do that.

File: var/spack/repos/builtin/packages/seqan/package.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    structures for the analysis of sequences with the focus on biological data.
+    Our library applies a unique generic design that guarantees high
+    performance, generality, extensibility, and integration with other
+    libraries. SeqAn is easy to use and simplifies the development of new
+    software tools with a minimal loss of performance"""
+
+    homepage = "https://www.seqan.de"
+    url      = "https://github.com/seqan/seqan/archive/seqan-v2.4.0.tar.gz"
+
+    version('2.4.0rc2', 'bd62772b3661bf07dad2bf8f69a1bdaf')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/seqan/package.py]: We generally don't add release candidates to Spack. Can you change this to the version and checksum for 2.4.0?

File: var/spack/repos/builtin/packages/seqan/package.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    depends_on('cmake@3.4.0:', type='build')
+    depends_on('python@2.7.0:', type='build')
+    depends_on('py-nose')
+    depends_on('py-sphinx')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/seqan/package.py]: These 2 should probably by `type=('build', 'run')` or just `type='build'` depending on whether they are actually used at runtime.

File: var/spack/repos/builtin/packages/seqan/package.py (hunk 2 above)
Comment (Sinan81) [reply var/spack/repos/builtin/packages/seqan/package.py]: sure.

File: var/spack/repos/builtin/packages/seqan/package.py (hunk 3 above)
Comment (Sinan81) [reply var/spack/repos/builtin/packages/seqan/package.py]: I think they are just build type as it seems python & sphinx is used for generating documentation.

**Pull request merged**